### PR TITLE
Support for multi-robot and different global planners

### PR DESCRIPTION
**Pull request by hawesie**
This brings in changes we've been using internally for multi-robot support and for running with different navigation planners.

### CODE REVIEW

**Comment by hawesie** [comment]
I'm still testing this locally, but tagging @bfalacerda @Jailander for info

**Comment by hawesie** [comment]
This works locally, so I'd like it to be merged if possible. I don't know how it effects other open tasks though.

File: emergency_behaviours/scripts/safety_trigger.py
Comment (marc-hanheide) [inline emergency_behaviours/scripts/safety_trigger.py]: This is a typo!

File: topological_navigation/src/topological_navigation/policies.py
Comment (marc-hanheide) [inline topological_navigation/src/topological_navigation/policies.py]: in general, would be better to define defaults for new parameters to keep the API backwards-compatible if possible, e.g. `color='red'`.

**Review by marc-hanheide — DISMISSED** [review]
I want to flag this up to @Jailander and @francescodelduchetto as the side effect of these changes is that if we run topo nav in some namespace (we may have done so other reasons than to support multiple robots) it'll break a system config very likely. There are a few changes I requested, but otherwise it looks fine for me. But we need all be aware that this can be the cause of future problems.

**Comment by hawesie** [comment]
@marc-hanheide Fixed those issues. Thanks for the review! I will wait for a :+1: from @francesodelduchetto or @Jailander before merging

**Review by marc-hanheide — APPROVED** [review]
I'm happy now, but @Jailander should still confirm, please.

**Comment by Jailander** [comment]
Hi yes just checked it I am happy

**Comment by hawesie** [comment]
Cool, thanks.

**Comment by marc-hanheide** [comment]
And... as expected there are side effects. We just spent a lot of time drilling down to the problem of `localise_by_topic` not working anymore. Look at this blame: https://github.com/strands-project/strands_navigation/blame/indigo-devel/topological_navigation/scripts/localisation.py#L30

The problem is that the topological map definition contained

```
localise_by_topic: '"{\"topic\": \"/battery_state\", \"field\": \"charging\",
      \"val\": true, \"localise_anywhere\": false}"'
```

So, with the new code this became `//battery_state` and didn't work anymore. Pinging @Jailander and @hawesie to make aware. Solution is to change it in the definition of the topomap to `battery_state`.

**Pull request merged**
